src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -835,8 +835,8 @@ object ICustomAstVisitor.VisitBinaryExpression(BinaryExpressionAst binaryExpress
             }
 
             List<PSTypeName> rhsTypes = InferTypes(binaryExpressionAst.Right).ToList();
-            var addedReturnTypes = new HashSet<string>();
-            var result = new List<PSTypeName>();
+            HashSet<string> addedReturnTypes = [];
+            List<PSTypeName> result = [];
```

Comment:
I'd prefer classic style.
(I'm not against the new syntax, but only if it's used everywhere. We can do this later by using the analyzer auto fix.)
```suggestion
            HashSet<string> addedReturnTypes = new HashSet<string>();
            List<PSTypeName> result = new List<PSTypeName>();
```

Reply:
I also prefer the old style, but with `var` so I don't end up with the same typename on both sides of the assignment.

Reply:
My initial request was about `foreach (var lType in lhsTypes)` :smile:

Reply:
I missed that. I've added the proper type here.